pkg/internal/nodeinfo/cpu_models.go

```diff
@@ -0,0 +1,333 @@
+package nodeinfo
+
+import (
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/kubevirt/hyperconverged-cluster-operator/api/v1beta1"
+)
+
+const (
+	// CPU model labels prefix used by KubeVirt
+	CpuModelLabelPrefix = "cpu-model.node.kubevirt.io/"
+)
+
+// CPU model PassMark scores mapping
+var cpuModelPassMarkScores = map[string]int{
+	// Intel CPU Models
+	"Raptor Lake":          58800,
+	"Alder Lake":           34400,
+	"Rocket Lake":          24400,
+	"Tiger Lake":           15000,
+	"Ice Lake":             12000,
+	"Comet Lake":           18200,
+	"Cascade Lake":         22000,
+	"Coffee Lake":          13600,
+	"Kaby Lake":            9600,
+	"Skylake":              8900,
+	"Skylake-Client":       8900,
+	"Skylake-Client-IBRS":  8900,
+	"Skylake-Server":       15000,
+	"Skylake-Server-IBRS":  15000,
+	"Broadwell":            7800,
+	"Broadwell-IBRS":       7800,
+	"Broadwell-noTSX":      7800,
+	"Broadwell-noTSX-IBRS": 7800,
+	"Haswell":              7200,
+	"Haswell-IBRS":         7200,
+	"Haswell-noTSX":        7200,
+	"Haswell-noTSX-IBRS":   7200,
+	"IvyBridge":            6400,
+	"IvyBridge-IBRS":       6400,
+	"SandyBridge":          5600,
+	"SandyBridge-IBRS":     5600,
+	"Westmere":             4200,
+	"Westmere-IBRS":        4200,
+	"Nehalem":              3800,
+	"Nehalem-IBRS":         3800,
+	"Penryn":               2400,
+	"Conroe":               1800,
+
+	// AMD CPU Models
+	"Zen 4":      62500,
+	"Zen 3":      39000,
+	"Zen 2":      32500,
+	"Zen+":       18100,
+	"Zen":        17500,
+	"Piledriver": 8500,
+	"Bulldozer":  7200,
+	"K10":        3200,
+	"K8":         1200,
+
+	// Generic/Common Models
+	"EPYC":       25000,
+	"EPYC-IBPB":  25000,
+	"EPYC-Rome":  35000,
+	"EPYC-Milan": 45000,
+	"EPYC-Genoa": 55000,
+	"Opteron_G1": 800,
+	"Opteron_G2": 1000,
+	"Opteron_G3": 1400,
+	"Opteron_G4": 4200,
+	"Opteron_G5": 6800,
+}
+
+var (
+	cpuModelInfo = &CpuModelInfo{
+		models:         make(map[string]*v1beta1.CpuModelInfo),
+		lock:           &sync.RWMutex{},
+		lastUpdateHash: "",
+	}
+)
+
+type CpuModelInfo struct {
+	models         map[string]*v1beta1.CpuModelInfo
+	lock           *sync.RWMutex
+	lastUpdateHash string
+	// Cluster totals for normalization
+	totalNodes  int
+	totalCpu    float64
+	totalMemory float64
+}
+
+func GetRecommendedCpuModels() []v1beta1.CpuModelInfo {
```

Comment:
This is a relatively pricey function, but it called for each reconcile loop, while the input for it only modified upon a node event. Can we run its logic as part of `processCpuModels`, and this function will only return the cached result?